crates/ide-assists/src/handlers/generate_getter.rs

```diff
@@ -72,88 +75,263 @@ pub(crate) fn generate_getter_mut(acc: &mut Assists, ctx: &AssistContext<'_>) ->
+        if parent_struct.is_none() {
+            return None;
+        }
+
+        let (info_of_record_fields, field_names) =
+            extract_and_parse_record_fields(&node, ctx.selection_trimmed(), mutable)?;
+
+        (parent_struct.unwrap(), info_of_record_fields, field_names)
+    } else {
+        // Single Record Field mode
+        let strukt = ctx.find_node_at_offset::<ast::Struct>()?;
+        let field = ctx.find_node_at_offset::<ast::RecordField>()?;
 
-    let field_name = field.name()?;
-    let field_ty = field.ty()?;
+        let record_field_info = parse_record_field(field, mutable)?;
 
-    // Return early if we've found an existing fn
-    let mut fn_name = to_lower_snake_case(&field_name.to_string());
-    if mutable {
-        format_to!(fn_name, "_mut");
+        let fn_name = record_field_info.fn_name.clone();
+
+        (strukt, vec![record_field_info], vec![fn_name])
+    };
+
+    // No record fields to do work on :(
+    if info_of_record_fields.len() == 0 {
+        return None;
     }
-    let impl_def = find_struct_impl(ctx, &ast::Adt::Struct(strukt.clone()), fn_name.as_str())?;
+
+    // `find_struct_impl` looks for impl of a struct, but this also has additional feature
+    // where it takes a list of function names and check if they exist inside impl_, if
+    // even one match is found, it returns None
+    let impl_def = find_struct_impl(ctx, &ast::Adt::Struct(strukt.clone()), fn_names)?;
 
     let (id, label) = if mutable {
         ("generate_getter_mut", "Generate a mut getter method")
     } else {
         ("generate_getter", "Generate a getter method")
     };
-    let target = field.syntax().text_range();
+
+    let mut target: TextRange = info_of_record_fields[0].target;
+
+    // Computing collective text range of all record fields in selected region
+    info_of_record_fields
+        .clone()
+        .iter()
+        .for_each(|record_field_info| target = target.cover(record_field_info.target));
+
+    let getter_info = GetterInfo { impl_def, strukt, mutable };
+
     acc.add_group(
         &GroupLabel("Generate getter/setter".to_owned()),
         AssistId(id, AssistKind::Generate),
         label,
         target,
         |builder| {
+            let record_fields_count = info_of_record_fields.len();
+
             let mut buf = String::with_capacity(512);
 
-            if impl_def.is_some() {
-                buf.push('\n');
+            // Check if an impl exists
+            if let Some(ref impl_def) = getter_info.impl_def {
+                // Check if impl is empty
+                if let Some(assoc_item_list) = impl_def.assoc_item_list() {
+                    if assoc_item_list.assoc_items().next().is_some() {
+                        // If not empty then only insert a new line
+                        buf.push('\n');
+                    }
+                }
+            }
+
+            for i in 0..record_fields_count {
```

Comment:
```suggestion
            for (i, record_field_info) in info_of_record_fields.iter().enumerate() {
```
this is more idiomatic as we no longer need to idnex into `info_of_record_fields`

Reply:
Nice observation!